src/unwinder/arm/unwinder_arm.cpp

```diff
@@ -1467,3 +1467,58 @@ BOOL DacUnwindStackFrame(T_CONTEXT *pContext, T_KNONVOLATILE_CONTEXT_POINTERS* p
     
     return res;
 }
+
+// This function returns the length of a function using the new unwind info on arm.
+// Taken from minkernel\ntos\rtl\arm\ntrtlarm.h.
+ULONG
+RtlpGetFunctionEndAddress (
+    __in PRUNTIME_FUNCTION FunctionEntry,
+    __in ULONG ImageBase
+    )
+{
+    ULONG FunctionLength;
+    
+    FunctionLength = FunctionEntry->UnwindData;
+    if ((FunctionLength & 3) != 0) {
+        FunctionLength = (FunctionLength >> 2) & 0x7ff;
+    } else {
+        FunctionLength = *(PTR_ULONG)(ImageBase + FunctionLength) & 0x3ffff;
+    }
+    
+    return FunctionEntry->BeginAddress + 2 * FunctionLength;
```

Comment:
Did you check this with LLVM style unwind info on ARM?  I'm not sure if windows is ABI compatible.